client/landing/gutenboarding/components/domain-picker-popover/index.tsx

```diff
@@ -4,27 +4,43 @@
 import * as React from 'react';
 import { Popover } from '@wordpress/components';
 import { useViewportMatch } from '@wordpress/compose';
+import { select } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import DomainPicker, { Props as DomainPickerProps } from '@automattic/domain-picker';
+import { recordEnterModal, recordCloseModal } from '../../lib/analytics';
+import { STORE_KEY } from '../../stores/onboard';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-interface Props extends Omit< DomainPickerProps, 'tracksName' > {
+interface Props extends DomainPickerProps {
 	isOpen: boolean;
 }
 
-const DomainPickerPopover: React.FunctionComponent< Props > = ( { isOpen, ...props } ) => {
-	const onClose = props.onClose;
-
+const DomainPickerPopover: React.FunctionComponent< Props > = ( { isOpen, onClose, ...props } ) => {
 	// Popover expands at medium viewport width
 	const isMobile = useViewportMatch( 'medium', '<' );
 
+	const tracksName = 'DomainPickerPopover';
+
+	React.useEffect( () => {
+		if ( isOpen ) {
+			recordEnterModal( tracksName );
+		}
+	}, [ isOpen ] );
+
+	const handleClose = () => {
+		recordCloseModal( tracksName, {
+			selected_domain: select( STORE_KEY ).getSelectedDomain()?.domain_name,
+		} );
+		onClose && onClose();
```

Comment:
Nit again, this could be:

```suggestion
		onClose?.();
```

Reply:
Done.